serde_with/src/schemars_0_8.rs

```diff
@@ -362,6 +365,55 @@ impl<T> JsonSchemaAs<T> for DisplayFromStr {
     forward_schema!(String);
 }
 
+impl JsonSchemaAs<bool> for BoolFromInt<Strict> {
+    fn schema_name() -> String {
+        "BoolFromInt<Strict>".into()
+    }
+
+    fn schema_id() -> Cow<'static, str> {
+        "serde_with::BoolFromInt<Strict>".into()
+    }
+
+    fn json_schema(_: &mut SchemaGenerator) -> Schema {
+        SchemaObject {
+            instance_type: Some(InstanceType::Integer.into()),
+            number: Some(Box::new(NumberValidation {
```

Comment:
I think this would be better modelled as an enum with the two values 0 and 1. A number with min and max would allow the value 0.5, I think.

Reply:
Normally you would be right, but the `instanceType: "integer"` part means that 0.5 would not be valid.

This is a good test case to have in case it gets modified in the future though. I have gone and added some tests to that effect.

Reply:
Yes, you are right :) 0.5 is not an integer